### PR TITLE
DM-13374: Deconstruct Butler prototype for redesign

### PR DESCRIPTION
**Pull request by pschella**

### CODE REVIEW

**Comment by timj** [comment]
I wonder how I missed those flake8 fixes.

**Comment by pschella** [comment]
Probably because they were the result of some other changes on this ticket (I intend to squash all commits).

File: python/lsst/daf/butler/core/datasets.py [hunk 1]
```diff
@@ -33,7 +32,7 @@ def _safeMakeMappingProxyType(data):
     return MappingProxyType(data)
 
 
-class DatasetType:
+class DatasetType(object):
```
Comment (timj) [inline python/lsst/daf/butler/core/datasets.py:35]: Is this needed? Python 3 doesn't need the explicit `object` inheritance.

**Review by timj — APPROVED** [review]
Looks fine (I'm trying not to worry yet about code that's disappearing). Thanks for cleaning up the `type` variable name confusion.

**Pull request merged**